site/pages/api/publish.api.ts

```diff
@@ -212,13 +213,29 @@ const publish = async (req: NextApiRequest, res: NextApiResponse) => {
 
         const draftDisruption = await getDisruptionById(validatedBody.data.disruptionId);
 
-        if (!draftDisruption) {
+        if (!draftDisruption || (draftDisruption && Object.keys(draftDisruption).length === 0)) {
             logger.error(`Disruption ${validatedBody.data.disruptionId} not found to publish`);
             redirectTo(res, ERROR_PATH);
 
             return;
         }
 
+        const validatedDisruptionBody = publishDisruptionSchema.safeParse(draftDisruption);
+
+        if (!validatedDisruptionBody.success) {
+            setCookieOnResponseObject(
+                COOKIES_REVIEW_DISRUPTION_ERRORS,
+                JSON.stringify({
```

Comment:
The structure of the error cookie message could follow `PageState<Partial<ReviewDisruptionProps>>` which would avoid type assertion in review page when the error and disruption is passed to the page